DragaliaAPI/Features/SavefileUpdate/V13Update.cs

```diff
@@ -0,0 +1,42 @@
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Emblem;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset;
+
+namespace DragaliaAPI.Features.SavefileUpdate;
+
+public class V13Update : ISavefileUpdate
+{
+    private readonly IStoryRepository storyRepository;
+    private readonly IEmblemRepository emblemRepository;
+    private readonly ILogger<V13Update> logger;
+
+    public V13Update(
+        IStoryRepository storyRepository,
+        IEmblemRepository emblemRepository,
+        ILogger<V13Update> logger
+    )
+    {
+        this.storyRepository = storyRepository;
+        this.emblemRepository = emblemRepository;
+        this.logger = logger;
+    }
+
+    public int SavefileVersion => 13;
+
+    public async Task Apply()
+    {
+        int[] readStoryIdList = storyRepository.UnitStories.Where(x => (x.State == StoryState.Read) && x.StoryType == StoryTypes.Chara).Select(x => x.StoryId).ToArray();
```

Comment:
```suggestion
        int[] readStoryIdList = await storyRepository.UnitStories.Where(x => (x.State == StoryState.Read) && x.StoryType == StoryTypes.Chara).Select(x => x.StoryId).ToArrayAsync();
```

Reply:
Would EF be able to compile a `.Where(x => x.Id % 10 == 5)`? Looks like all the story ids end in 1, 2, 3, 4, or 5.

Reply:
doubt it imo, I think an optimization would be to just precalculate all last chara stories then query based on that array/list + state

Reply:
Is it really that worth it to optimize here ? Given that it's only performed once per save file